src/tsserver/nodeServer.ts

```diff
@@ -715,6 +715,38 @@ namespace ts.server {
                 this.constructed = true;
             }
 
+            public send(msg: protocol.Message) {
+                if (useNodeIpc) {
+                    if (msg.type === "event" && !this.canUseEvents) {
+                        if (this.logger.hasLevel(LogLevel.verbose)) {
+                            this.logger.info(`Session does not support events: ignored event: ${JSON.stringify(msg)}`);
+                        }
+                        return;
+                    }

```

Comment:
This bit is probably not required; `IOSession` always passes true for `canUseEvents`.

Reply:
(I say that, knowing that there's code in this class that also checks it, probably unnecessarily as well.)

Reply:
Probably a good idea to gate the check anyway if there's a theoretical subclass.